internal/configconverter/discovery.go

```diff
@@ -77,7 +77,10 @@ func getServiceExtensions(out map[string]any) (map[string]any, []any, error) {
 	service := map[string]any{}
 	var serviceExtensions []any
 	if s, hasService := out["service"]; hasService && s != nil {
-		service = s.(map[string]any)
+		var ok bool
+		if service, ok = s.(map[string]any); !ok {
```

Comment:
Though not directly related to the fix, this issue was revealed during testing and is on the same path.